### DISCUSSION TITLE
Use of field: Changed Fields On Child Object

### DISCUSSION THREAD

**Discussion by baobao917** [jamessimone/apex-rollup#508] [Q&A] [answered]
The following is the excerpt from the README on this field:

> Changed Fields On Child Object (optional) - comma-separated list of field API Names to filter items from being used in the rollup calculations unless all the stipulated fields have changed. If you are using a Child Object Where Clause, I would not recommend using this field.> 

1. The text confused me where it stated "filter items...".  Does "Item" mean "rollup"?  I interpreted this to mean that a rollup wouldn't calculate unless the child object field(s) entered in this CMDT field all changed.  Is that correct?

2. If the above is correct, then would this work for formula fields? (i.e a change to a formula field such as Total Cost/Price)

**Comment by jamessimone** [comment]
Filter items refer to children records being updated. The point of this field is to allow you to avoid triggering the async part of a rollup if none of the fields contained therein have changed; it's like a where clause that states "only run when a record is new, has changes to (these fields) or is deleted."

I don't believe formula fields getting updated end up firing triggers, though

**Comment by baobao917** [comment, marked as answer]
@jamessimone Thanks.  Saw in v1.5.88 that you changed this README info.  It now reads as follows:
![image](https://github.com/jamessimone/apex-rollup/assets/104730171/bd6cddeb-ce2a-48b5-b0b7-a19315d86e1a)

- I noticed that the prior README txt stated that ALL the stipulated fields would have to change in order to trigger a rollup (see prior copy/paste excerpt in this thread).  In the amended version, it's stating that ANY of the stipulated fields would have to change in order to trigger a rollup.  Is the latter (ANY) txt a change in functionality or just a correction of incorrect prior documentation?

**Reply by jamessimone** [reply]
@baobao917 good catch - I was updating the verbiage from ALL to ANY because it's _always_ been ANY field changing (since [v1.0.9](https://github.com/jamessimone/apex-rollup/releases/tag/v1.0.9), at least!) and thanks to your discussion here I realized the README had been wrong for years 😅
